File: Day5/solution.js
//BFFFBBFRRR

const fs = require('fs');
const tickets = fs.readFileSync('./input.txt', 'utf-8').split('\r\n');


function getSeatID(str) {
    let firstPart = str.substr(0, 7).replace(/B/g, '1').replace(/F/g, '0');
    let secondPart = str.substr(7).replace(/R/g, '1').replace(/L/g, '0')

    let row = parseInt(firstPart, 2)
    let column = parseInt(secondPart, 2);

    return row * 8 + column    
}


// Part 1 - Highest ID
let highestID = tickets.reduce((a,c) => {
    let seatID = getSeatID(c)
     return seatID > a ? seatID : a
}, 0)

console.log('Part 1: ', highestID)

File: Day5/solution.py
import re


def parse_ticket(ticket):
    ticket = re.sub(r'[FL]', '0', ticket)
    ticket = re.sub(r'[BR]', '1', ticket)

    row = ticket[:7]
    seat = ticket[7:]

    return int(row, 2) * 8 + int(seat, 2)


data = [parse_ticket(x) for x in open('./input.txt').read().splitlines()]

print(f'Part 1: {max(data)}')

for seat_id in data:
    if not seat_id + 1 in data and seat_id + 2 in data:
        print(f'Part 2: {seat_id + 1}')
        break
